src/gpuarray/buffer.h

```diff
@@ -757,6 +763,48 @@ typedef enum _ga_usefl {
   GA_USE_OPENCL =   0x4000,
 } ga_usefl;
 
+/************************************************************************************
+*                         multi-gpu communicator interface                         *
+************************************************************************************/
+
+/**
+ * \brief Creates a new communicator (multi process version)
+ * \param clique_id [const char*] TODO
+ * \param ndev [int] TODO
+ * \param rank [int] TODO
+ * \param res [int*] TODO
+ * \return gpucomm* TODO
+ */
+GPUARRAY_PUBLIC gpucomm* gpucomm_init(const char* clique_id, int ndev, int rank,
+                                      int* res);
+
+/**
+ * \brief Frees resources associated with communicator object.
+ * \param comm [gpucomm*] TODO
+ * \return void TODO
+ */
+GPUARRAY_PUBLIC void gpucomm_destroy(gpucomm* comm);
+
+/**
+ * \brief Returns nice error message.
+ * \param comm [gpucomm*] TODO
+ * \param err [int] TODO
+ * \return const char* TODO
+ */
+GPUARRAY_PUBLIC const char* gpucomm_error(gpucomm* comm, int err);
+
+// TODO cuda extension:
+// 1. ncclGetUniqueId
+// 2. ncclCommCount (below mby ops in gpucomm struct?!)
+// 3. ncclCommUserRank
+// 4. ncclCommCuDevice
```

Comment:
:+1: 
`ncclGetUniqueId`, which is a helper function independent with gpus, shouldn't this suit more a as cuda extension? I mean it's not a gpu-related functionality.